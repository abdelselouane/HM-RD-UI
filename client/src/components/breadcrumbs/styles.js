import styled from 'styled-components';

export const Container = styled.div`
  grid-area: cmess;

  .breadcrumb {
    font-size: 16px;
  }
`;

export default Container;
