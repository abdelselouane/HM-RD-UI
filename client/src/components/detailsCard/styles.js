import styled from 'styled-components';

export const Card = styled.div`
  width: 85%;
  margin-bottom: 20px;

  .card-content {
    padding: 0px 25px;
  }
`;

export const Title = styled.span`
  align-items: flex-end;
  display: flex;
  justify-content: flex-start;
  width: 100%;
  border-bottom: solid #666 1px;
  padding-bottom: 15px;

  h2 {
    display: inline-block;
    color: #f96302;
    font-weight: 900;
    padding-right: 25px;
  }

  span {
    text-transform: capitalize;
  }
`;

export const Row = styled.div`
  display: flex;
  padding: 15px;
`;

export const Col = styled.div`
  ${props => props.flex && `flex: ${props.flex}`}
  display: flex;
  flex-direction: column;
`;

export const Label = styled.div`
  font-weight: 700;
  color: #666;
`;
