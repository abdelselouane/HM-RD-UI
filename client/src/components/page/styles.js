import styled, { css } from 'styled-components';

const defaultGrid = css`
  display: grid;
  grid-template-rows: 70px
                      48px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 72px);
  grid-template-columns: 20px 1fr repeat(4, 3fr) 1fr 20px;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . . . . . . .";
`;

const lowResolutionGrid = css`
  display: grid;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". . . . . . . .";
  grid-template-rows: 70px
                      24px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 48px);
  grid-template-columns: 10px 1fr repeat(4, 3fr) 1fr 10px;
`;

export const GridLayout = styled.div`
  background-color: ${({ theme }) => theme.black12};
  box-sizing: border-box;
  color: ${({ theme }) => theme.blackA11};
  height: 100%;  
  ${props => (props.enableScroll ? 'overflow-x: scroll;' : 'overflow: hidden;')}
  
  ${() => defaultGrid}
  @media only screen and (max-width: 1024px) {
    ${() => lowResolutionGrid}
  }
`;

export const Container = styled.div`
  align-items: flex-start;
  display: flex;
  grid-area: carea;
  justify-content: center;
  ${({ hideBoxShadow }) => (hideBoxShadow ? '' : 'box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12), 0 2px 2px 0 rgba(0, 0, 0, 0.24);')}
`;
