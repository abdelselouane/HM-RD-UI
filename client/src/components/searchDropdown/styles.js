import styled from 'styled-components';

const StyledSearchDropdown = styled.div`
  display: block;
  width: 700px;
  position: absolute;
  top: 70%;
  box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12), 0 2px 2px 0 rgba(0, 0, 0, 0.24);
  z-index: 5;

  .scrollable-table {
    overflow-y: hidden;
  }

  .standard-table tbody tr {
    td {
      background-color: ${({ theme }) => theme.white} !important;
    }

    &:hover td {
      background-color: ${({ theme }) => theme.black10} !important;
    }
  }

  .table-body.alternating-rows {
    width: 100%;
    
    tr:nth-child(2n + 1),
    tr:nth-child(2n) {
      background-color: ${({ theme }) => theme.black10};
    }

    tr:hover td,
    tr:focus td {
      background-color: ${({ theme }) => theme.black12} !important;
    }
    
    tr td {
      border: none;
    }
  }
`;

export default StyledSearchDropdown;
