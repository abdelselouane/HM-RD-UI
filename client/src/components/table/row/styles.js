import styled from 'styled-components';

const StyledTd = styled.td`
  ${props => {
    if (props.tdProps
      && props.tdProps.color
      && Object.keys(props.theme).includes(props.tdProps.color)) {
      return {
        ...props.tdProps,
        color: `${props.theme[props.tdProps.color]} !important`
      };
    }
    if (props.disableRow) {
      return {
        ...props.tdProps,
        color: `${props.theme.black14} !important`
      };
    }
    return props.tdProps;
  }}
`;

export default StyledTd;
