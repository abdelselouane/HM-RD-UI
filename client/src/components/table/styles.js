import styled, { css } from 'styled-components';

const expandableTableCss = css`
  &.standard-table > thead > tr > th {
    background-color: ${({ theme }) => theme.white} !important; 
    &.sortable {
      &:after {
        position: static !important;
        padding-left: 10px;
      }
    }  
  }

  &.standard-table > tbody > tr {
    > td {
      border-bottom: 1px solid ${({ theme }) => theme.black14};
    }

    &:nth-child(4n + 1):hover > td,
    &:nth-child(4n + 2):hover > td,
    &:nth-child(4n + 3):hover > td,
    &:nth-child(4n + 4):hover > td {
      background-color: unset !important;
    }

    &:nth-child(4n + 1) td,
    &:nth-child(4n + 2) td {
      background-color: ${({ theme }) => theme.black11} !important;
    }

    &:nth-child(4n + 3) td,
    &:nth-child(4n + 4) td {
      background-color: ${({ theme }) => theme.white} !important;
    }

    &[id^='expandable'] {
      &:nth-child(4n + 1):hover,
      &:nth-child(4n + 1):hover > td,
      &:nth-child(4n + 2):hover,
      &:nth-child(4n + 2):hover > td {
        background-color: ${({ theme }) => theme.black11} !important;
      }

      &:nth-child(4n + 3):hover,
      &:nth-child(4n + 3):hover > td,
      &:nth-child(4n + 4):hover,
      &:nth-child(4n + 4):hover > td {
        background-color: ${({ theme }) => theme.white} !important;
      }
    }
  }
`;

const clearOnHoverCss = css`
  &.standard-table > thead > tr:hover,
  &.standard-table > thead > tr:hover > th {
    background-color: inherit !important;
  }
`;

export const StyledTable = styled.table`
  background-color: unset !important;
  ${({ isExpandable }) => isExpandable && expandableTableCss} ${({ clearOnHover }) =>
  clearOnHover && clearOnHoverCss};
`;

export const StyledTh = styled.th`
  ${props => {
    if (
      props.thProps &&
      props.thProps.color &&
      Object.keys(props.theme).includes(props.thProps.color)
    ) {
      return {
        ...props.thProps,
        color: `${props.theme[props.thProps.color]} !important`
      };
    }
    return props.thProps;
  }};
`;
