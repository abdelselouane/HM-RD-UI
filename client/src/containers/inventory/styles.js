import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
background-color: ${({ theme }) => theme.white};
box-sizing: border-box;
display: flex;
flex-direction: column;
justify-content: flex-start;
height: 100%;
overflow-y: hidden;
width: 100%;


.inventory-tabs {
  height: 100%;
}

.tabs span {
  &:focus {
    outline: none !important;
  }
  
  &[role=tab]:first-child {
    border-left: none;
  }
}

[role=tabpanel] {
    display: flex;
    flex-direction: column;
    height: 95%;
    overflow: hidden;
    padding: 20px 0px;

    .text-input-container.search {
      margin: 15px 15px;
      width: 65%;

      &.dropdown {
        width: 75%;
        margin: 15px 0px;
      }
    }
}

.search-container{
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  button {
    flex: 0.15;
  }
}

.text-input-container .input-field {
  margin-top: 0em;
}

.edit-col .button {
  margin: 0px;
}

.text-input-container.search {
  flex: 0.5;

  input {
    margin-top: -6px;
  }

  .icon_search {
    bottom: 0;
    color: ${({ theme }) => theme.orange};
    font-size: 30px;
    padding: 5px 10px;
    position: absolute;
    right: 0;
    z-index: 1;
  }
}
`;

export default Container;
