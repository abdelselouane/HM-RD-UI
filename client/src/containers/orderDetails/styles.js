import styled from 'styled-components';
import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
  background-color: ${({ theme }) => theme.white};
  align-items: flex-start;
  display: flex;
  flex-direction: column;
`;

export const CardTitle = styled.div`
  align-items: flex-end;
  display: flex;
  justify-content: space-between;
  height: 50px;
  margin: 0px;
  margin-bottom: 30px;
  padding: 0px 20px;
  width: 100%;

  h2 {
    display: inline-block;
    margin-bottom: 0px;
  }
`;

export const PurchaseOrderList = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  padding-top: 20px;
  flex: 1;
  overflow-y: scroll;
  overflow-x: hidden;
  width: 100%;
`;
