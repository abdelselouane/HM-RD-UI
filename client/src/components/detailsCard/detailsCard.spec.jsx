/* global describe, beforeAll, it, expect */
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import { Card, Row, Col, Title, Label } from './styles';

import DetailsCard from './index';

describe('<DetailsCard />', () => {
  let wrapper;
  let props;

  beforeAll(() => {
    props = {
      purchaseOrderNbr: '971070001',
      purchaseOrderStatDesc: 'SUBMITTED',
      parts: [
        {
          partDescription: 'mockDescription3',
          partNbr: 'abc-123',
          brandName: 'mockbrand',
          quantity: 1
        },
        {
          partDescription: 'mockDescription4',
          partNbr: 'cba-321',
          brandName: 'mockbrand',
          quantity: 2
        }
      ]
    };

    wrapper = shallow(<DetailsCard {...props} />);
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render styles properly', () => {
    [Card, Row, Col, Title, Label].forEach(Component => {
      expect(toJson(shallow(<Component flex="0" />))).toMatchSnapshot();
    });
  });
});
