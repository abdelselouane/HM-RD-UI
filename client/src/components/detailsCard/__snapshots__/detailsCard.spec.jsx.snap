// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsCard /> should render properly 1`] = `
<styled.div
  className="card"
>
  <div
    className="card-toolbar"
  >
    <styled.span
      className="card-title"
    >
      <h2>
        Purchase Order #971070001
      </h2>
      <span>
        Status: submitted
      </span>
    </styled.span>
  </div>
  <div
    className="card-content"
  >
    <styled.div
      className="part-details"
      key="abc-123"
    >
      <styled.div
        flex="0.4"
        key="partDescription"
      >
        mockDescription3
      </styled.div>
      <styled.div
        flex="0.2"
        key="brandName"
      >
        <styled.div>
          Brand
        </styled.div>
        mockbrand
      </styled.div>
      <styled.div
        flex="0.2"
        key="partNbr"
      >
        <styled.div>
          Part Number
        </styled.div>
        abc-123
      </styled.div>
      <styled.div
        flex="0.2"
        key="quantity"
      >
        <styled.div>
          Quantity
        </styled.div>
        1
      </styled.div>
    </styled.div>
    <styled.div
      className="part-details"
      key="cba-321"
    >
      <styled.div
        flex="0.4"
        key="partDescription"
      >
        mockDescription4
      </styled.div>
      <styled.div
        flex="0.2"
        key="brandName"
      >
        <styled.div>
          Brand
        </styled.div>
        mockbrand
      </styled.div>
      <styled.div
        flex="0.2"
        key="partNbr"
      >
        <styled.div>
          Part Number
        </styled.div>
        cba-321
      </styled.div>
      <styled.div
        flex="0.2"
        key="quantity"
      >
        <styled.div>
          Quantity
        </styled.div>
        2
      </styled.div>
    </styled.div>
  </div>
</styled.div>
`;

exports[`<DetailsCard /> should render styles properly 1`] = `
.c0 {
  width: 85%;
  margin-bottom: 20px;
}

.c0 .card-content {
  padding: 0px 25px;
}

<div
  className="c0"
/>
`;

exports[`<DetailsCard /> should render styles properly 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 15px;
}

<div
  className="c0"
/>
`;

exports[`<DetailsCard /> should render styles properly 3`] = `
.c0 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`<DetailsCard /> should render styles properly 4`] = `
.c0 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
  border-bottom: solid #666 1px;
  padding-bottom: 15px;
}

.c0 h2 {
  display: inline-block;
  color: #f96302;
  font-weight: 900;
  padding-right: 25px;
}

.c0 span {
  text-transform: capitalize;
}

<span
  className="c0"
/>
`;

exports[`<DetailsCard /> should render styles properly 5`] = `
.c0 {
  font-weight: 700;
  color: #666;
}

<div
  className="c0"
/>
`;
